common/src/java/org/apache/hadoop/hive/ql/ErrorMsg.java

```diff
@@ -492,6 +492,7 @@ public enum ErrorMsg {
   NONICEBERG_COMPACTION_WITH_FILTER_NOT_SUPPORTED(10440, "Compaction with filter is not allowed on non-Iceberg table {0}.{1}", true),
   ICEBERG_COMPACTION_WITH_PART_SPEC_AND_FILTER_NOT_SUPPORTED(10441, "Compaction command with both partition spec and filter is not supported on Iceberg table {0}.{1}", true),
   COMPACTION_THREAD_INITIALIZATION(10442, "Compaction thread failed during initialization", false),
+  ALTER_TABLE_COMPACTION_NON_PARTITIONED_COLUMN_NOT_ALLOWED(10443, "Filter expression can contain only partition columns."),
```

Comment:
do we have test cases for partition evolution? say column `a` was in a list of partition columns, but after the partition evolution it was changed to column `b`
what happens if we use `a` in a filter predicate?

Reply:
Added test case in iceberg_major_compaction_partition_evolution_w_id_spec_w_filter.q.